torch/_inductor/fx_passes/mkldnn_fusion.py

```diff
@@ -1146,17 +1146,30 @@ def linear(match, *args, **kwargs):
                     if has_free_symbols(batch_size)
                     else batch_size,
                 )
+                # MKL packed matrix can't be copied to a different address because the internal implementation
+                # depends on the alignment of internally-stored metadata.
+                # In aot mode, we need to firstly save the packed weight, when loading it,
+                # it will be in a different address which doesn't work.
+                # Disable MKL prepack linear in AOT mode
```

Comment:
Is there any plan to support this in future?

Reply:
Yes, the plan is to switch to oneDNN for FP32 linear as well to resolve this limitation of MKL.

Currently BF16 linear uses oneDNN (won't have this issue) while FP32 linear uses MKL since in several cases, MKL FP32 linear performs better than MKLDNN FP32 linear. We're collecting such cases where MKLDNN FP32 linear performs worse and will report to oneDNN for optimizations cc @chuanqi129 @zxd1997066. Once such performance gap is fixed in oneDNN, we'll switch FP32 linear to oneDNN for both JIT Inductor and AOTI.